src/main/java/com/limechain/utils/json/ObjectMapper.java

```diff
@@ -67,8 +68,12 @@ private <T> T convertValue(Class<T> type, Object value) {
             return (T) value;
         } else if (type == Integer.class || type == int.class) {
             return (T) (Integer) ((Number) value).intValue();
+        } else if (type == Long.class || type == long.class) {
+            return (T) (Long) ((Number) value).longValue();
         } else if (type == Double.class || type == double.class) {
             return (T) (Double) ((Number) value).doubleValue();
+        } else if (type == BigInteger.class) {
+            return (T) BigInteger.valueOf((Long) value);
```

Comment:
Doesn't seem very safe. What happens if value exceeds max long value? Isn't it better to cast to `String` and create a `BigInteger` from it?